### PR TITLE
[PM-10326] Hide 'remove' option for organization managed members

### PR DESCRIPTION
**Pull request by r-tome**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-10326

## 📔 Objective

Hide the ‘remove’ option on the members list and on the individual member dialog for managed members

## 📸 Screenshots

image

image

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: apps/web/src/app/admin-console/organizations/members/components/member-dialog/member-dialog.component.html [hunk 1]
```diff
@@ -262,7 +262,7 @@ <h3 class="tw-mt-4">
           {{ "revokeAccess" | i18n }}
         </button>
         <button
-          *ngIf="editMode"
+          *ngIf="editMode && params.managedByOrganization === false"
```
Comment (BTreston) [inline apps/web/src/app/admin-console/organizations/members/components/member-dialog/member-dialog.component.html:265]: ❓ Would this prevent you from removing the user form an org they are a member of where they are not a managed user?

Comment (r-tome) [reply apps/web/src/app/admin-console/organizations/members/components/member-dialog/member-dialog.component.html:265]: No, it only prevents removing a user who _is_ a managed user, as they have the **Delete** action instead. An alternative condition could be `*ngIf="editMode && params.managedByOrganization !== true`

**Review by BTreston — DISMISSED** [review]

File: apps/web/src/app/admin-console/organizations/members/components/member-dialog/member-dialog.component.html [hunk 2]
```diff
@@ -262,7 +262,11 @@ <h3 class="tw-mt-4">
           {{ "revokeAccess" | i18n }}
         </button>
         <button
-          *ngIf="editMode"
+          *ngIf="
+            editMode &&
+            ((accountDeprovisioningEnabled$ | async) === false ||
+              params.managedByOrganization === false)
```
Comment (BTreston) [inline apps/web/src/app/admin-console/organizations/members/components/member-dialog/member-dialog.component.html:268]: `accountDeprovisioningEnabled$ | async` and `manangedByOrganization' here will be booleans, I don't think we need the === check. We should be able to just use their values for these conditions:

```suggestion
            editMode &&
           (!(accountDeprovisioningEnabled$ | async) || !params.managedByOrganization)
```

File: apps/web/src/app/admin-console/organizations/members/components/member-dialog/member-dialog.component.html [hunk 3]
```diff
@@ -272,7 +276,11 @@ <h3 class="tw-mt-4">
           [disabled]="loading"
         ></button>
         <button
-          *ngIf="editMode && params.managedByOrganization === true"
+          *ngIf="
+            editMode &&
+            (accountDeprovisioningEnabled$ | async) === true &&
+            params.managedByOrganization === true
```
Comment (BTreston) [inline apps/web/src/app/admin-console/organizations/members/components/member-dialog/member-dialog.component.html:282]: Same comment as above for these

File: apps/web/src/app/admin-console/organizations/members/members.component.html [hunk 4]
```diff
@@ -315,13 +315,24 @@
...
-                <button type="button" bitMenuItem (click)="remove(u)">
+                <button
+                  *ngIf="
+                    (accountDeprovisioningEnabled$ | async) === false ||
+                    u.managedByOrganization === false
+                  "
+                  type="button"
+                  bitMenuItem
+                  (click)="remove(u)"
+                >
```
Comment (BTreston) [inline apps/web/src/app/admin-console/organizations/members/members.component.html:326]: This file too

**Review by BTreston — CHANGES_REQUESTED** [review]
Looks good, just some ⛏️ s to keep the logic clean

File: apps/web/src/app/admin-console/organizations/members/components/member-dialog/member-dialog.component.html (hunk 2 above)
Comment (r-tome) [reply apps/web/src/app/admin-console/organizations/members/components/member-dialog/member-dialog.component.html]: Makes sense, I've made that change across all conditions

**Review by BTreston — APPROVED** [review]

**Pull request merged**